exonum/src/blockchain/mod.rs

```diff
@@ -271,6 +271,11 @@ impl Blockchain {
                     .expect("Transaction not found in the database.");
             }
 
+            // Invoke execute method for all services.
+            self.service_map
+                .values()
+                .for_each(|service| service_execute(service.as_ref(), &mut fork));
```

Comment:
From `std` docs, it seems that `for` cycle could be preferable here.

Reply:
But `iter().for_each()` in many cases can be transparently replaced by `par_iter().for_each()` from the `rayon`.